### DISCUSSION TITLE
Re-route request to another destination if first fails in cluster

### DISCUSSION THREAD

**Discussion by Henery309** [dotnet/yarp#790] [Q&A] [answered]
Hi, 

Is it possible to re-route request to second destination if the first destination returns http 500?
With my setup the first destination can sometimes fail with http 500 error. We have two deployments for the same service and wanted to re-direct the request to second destination if first fails. 

```json
{
  "ReverseProxy": {
    "Routes": [
      {
        "RouteId": "app",
        "ClusterId": "app",
        "Match": {
          "Methods": [ "Get" ],
          "Path": "/app/v1"
        },
        "Transforms": [
          {
            "PathSet": "/weatherforecast"
          }
        ]
      }
    ],
    "Clusters": {
      "app": {
        "LoadBalancingPolicy": "First",
        "Destinations": {
          "cluster1/destination1": {
            "Address": "https://localhost:5001/"
          },
          "cluster1/destination2": {
            "Address": "https://localhost:5003/"
          }
        }
      }
    }
  }
}
```

**Comment by Tratcher** [comment, marked as answer]
Retries haven't been implemented, see https://github.com/microsoft/reverse-proxy/issues/56. Feel free to add requirements there. It should be possible to build basic retry support on today's system as a middleware if you're interested in collaborating on that?

**Reply by Henery309** [reply]
I am looking for a workaround for now as we are having issue in prod (1 of the service fails with http 500 and requires manual restart I want Yarp to start using second destination). I did little more digging for this and came across `IDestinationHealthUpdater` I'd like to mark the destination that had http 500 error as unhealthy from middleware. Is there a way to find which cluster and destination failed with http 500 error in middleware?

**Reply by Tratcher** [reply]
`IDestinationHealthUpdater` is part of the health checks infrastructure. Before you try calling it directly I suggest enabling passive health checks that should handle this for you: https://microsoft.github.io/reverse-proxy/articles/dests-health-checks.html#passive-health-checks
